pkg/server/image_remove.go

```diff
@@ -17,17 +17,62 @@ limitations under the License.
 package server
 
 import (
-	"golang.org/x/net/context"
+	"fmt"
 
+	"github.com/containerd/containerd/images"
+	"github.com/golang/glog"
+	"golang.org/x/net/context"
 	"k8s.io/kubernetes/pkg/kubelet/api/v1alpha1/runtime"
+
+	"github.com/kubernetes-incubator/cri-containerd/pkg/metadata"
 )
 
 // RemoveImage removes the image.
 // TODO(mikebrow): harden api
-func (c *criContainerdService) RemoveImage(ctx context.Context, r *runtime.RemoveImageRequest) (*runtime.RemoveImageResponse, error) {
-	// Only remove image from the internal metadata store for now.
-	// Note that the image must be digest here in current implementation.
-	// TODO(mikebrow): remove the image via containerd
-	err := c.imageMetadataStore.Delete(r.GetImage().GetImage())
-	return &runtime.RemoveImageResponse{}, err
+// TODO(random-liu): Update CRI to pass image reference instead of ImageSpec.
```

Comment:
Can link my issue if you want: https://github.com/kubernetes/kubernetes/issues/46255